packages/sd-jwt/src/action-handler.ts

```diff
@@ -70,28 +69,43 @@ export class SDJwtPlugin implements IAgentPlugin {
 
   /**
    * Get the key to sign the SD-JWT
-   * @param issuer - did url like did:exmaple.com#key-1
+   * @param identifier - identifier like a did and other forms of identifiers
    * @param context - agent instance
    * @returns the key to sign the SD-JWT
    */
-  private async getSignKey(issuer: string, context: IRequiredContext) {
-    debug(`Getting signing key for issuer ${issuer}`)
-    const identifier = await context.agent.didManagerGet({
-      did: issuer.split('#')[0],
-    })
-    const doc = await mapIdentifierKeysToDocWithJwkSupport({ identifier, vmRelationship: 'assertionMethod' }, context)
-    if (!doc || doc.length === 0) {
-      throw new Error('No key found for signing')
-    }
-    const key = doc.find((key) => key.meta.verificationMethod.id === issuer)
-    if (!key) {
-      throw new Error(`No key found with the given id: ${issuer}`)
+  async getSignKey (
+      identifier: string,
+      context: IRequiredContext
+      //todo make the type of key known
+  ): Promise<{ alg: string; key: any }> {
+    if (identifier.startsWith('did:')) {
+      const didDocument = await context.agent.resolveDid({ didUrl: identifier });
+      const key = didDocument.verificationMethod?.find(vm => vm.publicKeyJwk);
```

Comment:
This will not work. You need agent Keys not a VM.

We have getFirstKeyWithRelation method, that does the mapping of agent available keys onto a DID  Document.
In this case you would need an additional option to the method to define whether it is an authentication (holder) or assertionMethod (issuer)

Reply:
changed the bit that handles the did identifiers and used different functionalities to get to the key object